lang/syn/src/codegen/program/handlers.rs

```diff
@@ -694,6 +694,15 @@ pub fn generate(program: &Program) -> proc_macro2::TokenStream {
             let anchor = &ix.anchor_ident;
             let variant_arm = generate_ix_variant(ix.raw_method.sig.ident.to_string(), &ix.args);
             let ix_name_log = format!("Instruction: {}", ix_name);
+            let ret_type = &ix.returns.ty.to_token_stream();
+            let result_handler = match ret_type.to_string().as_str() {
+                "()" => quote! {},
+                _ => quote! {
+                    let mut buffer: Vec<u8> = Vec::new();
```

Comment:
wdyt about allowing users to return a `Result<Option<T>>`? Then instructions could be built in a way where the caller can specify via an argument whether they want the return value

Reply:
I think `Result<Something>` vs `Result<()>` might be enough for the moment at least. My hunch is that a function that only returns something some of the time is a bit of an edge case, and people will mostly want the function to return all the time. I could be wrong, but I'd propose we go with this approach and reevaluate based on the use cases we see.